### PR TITLE
add tests for php5.6

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
The new spec file alone is working. Our module doesn't support the upgrade/downgrade of the php version, and beaker-rspec executes all tests in the same container.

**Comment by ekohl** [comment]
If we do go this way, I'd prefer the following in ```.sync.yml```

```
---
.travis.yml:
  docker_sets:
    - set: docker/ubuntu-16.04
      options:
        script: 'bundle exec rspec spec/acceptance/php56_spec.rb'
    - set: docker/ubuntu-14.04
      options:
        script: 'bundle exec rspec spec/acceptance/php56_spec.rb'
    - set: docker/centos-7
      options:
        script: 'bundle exec rspec spec/acceptance/php56_spec.rb'
    - set: docker/debian-8
      options:
        script: 'bundle exec rspec spec/acceptance/php56_spec.rb'
    - set: docker/ubuntu-16.04
      options:
        script: 'bundle exec rspec spec/acceptance/php_spec.rb'
    - set: docker/ubuntu-14.04
      options:
        script: 'bundle exec rspec spec/acceptance/php_spec.rb'
    - set: docker/centos-7
      options:
        script: 'bundle exec rspec spec/acceptance/php_spec.rb'
    - set: docker/debian-8
      options:
        script: 'bundle exec rspec spec/acceptance/php_spec.rb'
```

File: spec/acceptance/php56_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        phpunit      => false,
+      }
+      EOS
+      # Run it twice and test for idempotency
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    case default[:platform]
+    when %r{16.04}
```
Comment (ekohl) [inline spec/acceptance/php56_spec.rb]: Isn't this the same as 14.04? Would it be better to test for the Debian OS family or at least Ubuntu?

File: spec/acceptance/php_spec.rb [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
     end
-  when %r{14.04}
-    describe package('php5-fpm') do
-      it { is_expected.to be_installed }
-    end
-  when %(7)
-    describe package('php-fpm') do
-      it { is_expected.to be_installed }
+
+    case default[:platform]
```
Comment (ekohl) [inline spec/acceptance/php_spec.rb]: I'd prefer to set the expected package name in a case statement (with a ```let``` block?) and then have the same ```describe``` block with a variable

Comment (bastelfreak) [reply spec/acceptance/php_spec.rb]: implemented it, just without the let block (didn't work)

File: spec/acceptance/php56_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+require 'spec_helper_acceptance'
+
+describe 'with specific php version' do
+  case default[:platform]
+  when %r{16.04}
```
Comment (ekohl) [inline spec/acceptance/php56_spec.rb]: Wouldn't it be better to just check for Ubuntu here?

File: spec/acceptance/php56_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    packagename = 'php5.6-fpm'
+  when %r{14.04}
+    packagename = 'php5.6-fpm'
+  when %r{7}
```
Comment (ekohl) [inline spec/acceptance/php56_spec.rb]: Is this Debian 7 or CentOS 7?

File: spec/acceptance/php56_spec.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    packagename = 'php-fpm'
+  when %r{8}
```
Comment (ekohl) [inline spec/acceptance/php56_spec.rb]: I'm guessing Debian?

File: spec/acceptance/php56_spec.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
+require 'spec_helper_acceptance'
+
+describe 'with specific php version' do
```
Comment (ekohl) [inline spec/acceptance/php56_spec.rb]: How about a check for ```php --version```?

File: spec/acceptance/php56_spec.rb (hunk 3 above)
Comment (bastelfreak) [reply spec/acceptance/php56_spec.rb]: yeah would work as well

File: spec/acceptance/php_spec.rb [hunk 7]
```diff
@@ -1,24 +1,25 @@
...
+    when %r{ubuntu-16.04}
+      package = 'php7.0-fpm'
+    when %r{ubuntu-14.04}
+      package = 'php5-fpm'
+    when %{centos}
+      package = 'php-fpm'
+    when %{'debian'}
```
Comment (ekohl) [inline spec/acceptance/php_spec.rb]: Are the quotes needed here? should it be a regex too?

File: spec/acceptance/php56_spec.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    describe package(packagename) do
+      it { is_expected.to be_installed }
+    end
+    describe service(packagename) do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+    describe command('php --version') do
+      its(:stdout) { is_expected.to match %r{5.6} }
```
Comment (ekohl) [inline spec/acceptance/php56_spec.rb]: Should the dot in 5.6 be escaped since it's a regex char?

**Comment by bastelfreak** [comment]
finally green

**Pull request merged**